modules/extension/wmts/src/main/java/org/geotools/data/wmts/model/TileMatrixSet.java

```diff
@@ -0,0 +1,131 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2017, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.wmts.model;
+
+import java.util.ArrayList;
+import org.geotools.data.ows.CRSEnvelope;
+
+import org.geotools.referencing.CRS;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.NoSuchAuthorityCodeException;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+
+
+public class TileMatrixSet {
+    private static final CoordinateReferenceSystem WEB_MERCATOR_CRS;
+
+    static {
+        CoordinateReferenceSystem tmpCrs = null;
+
+        try {
+            tmpCrs = CRS.decode("EPSG:3857");
+        } catch (FactoryException e) {
+            /*LOGGER.log(Level.SEVERE,
+                    "Failed to create Web Mercator CRS EPSG:3857", e);*/
+            throw new RuntimeException(e);
+        }
+        WEB_MERCATOR_CRS = tmpCrs;
+
+    }
+    final public static String OWS = "http://www.opengis.net/ows/1.1";
+
+    private String identifier;
+
+    private String crs;
+
+    private CRSEnvelope bbox;
+
+    private ArrayList<TileMatrix> matrices = new ArrayList<>();
+
+    public void setIdentifier(String textContent) {
+        this.identifier = textContent;
+
+    }
+
+    public void setCRS(String textContent) {
+        this.crs = textContent;
+
+    }
+
+    public void addMatrix(TileMatrix processTM) {
+        matrices.add(processTM);
+
+    }
+
+    /**
+     * @return the crs
+     */
+    public String getCrs() {
+        return crs;
+    }
+
+    public CoordinateReferenceSystem getCoordinateReferenceSystem()
+            throws NoSuchAuthorityCodeException, FactoryException {
+        
+        //TODO: Kill who ever is still using this hack!
```

Comment:
Well well now!
I fully agree we should stop using it, but GWC still defaults to it. A more constructive approach:

- In the comment state that it's compatibility with the legacy code used before EPSG recognized the web mercator CRS
- Go to GWC and suggest changing the default to a sane and standard EPSG:3857 instead